exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -791,6 +799,43 @@ public FileGroupScan clone(FileSelection selection) throws IOException {
   }
 
   @Override
+  public GroupScan applyLimit(long maxRecords) {
```

Comment:
I was thinking about that as well. Theoretically, it would be best to do a sort on record count and then binary search to the row group that has the closest number greater than the requested amount (too small means multiple files, larger files require more metadata reading/parsing. However, it kind of seems like premature optimization to me. Are you seeing lots of people with many small Parquet files? That generally seems counter to the Parquet design.